ledger/ledger-api-client/src/test/suite/scala/com/digitalasset/ledger/client/services/commands/CommandTrackerFlowTest.scala

```diff
@@ -300,16 +280,51 @@ class CommandTrackerFlowTest
       }
 
       "timeout the command when the MRT passes" in {
+        val Handle(submissions, results, _, _) = runCommandTrackingFlow(allSubmissionsSuccessful)
+
+        submissions.sendNext(newSubmission(commandId, dedupTime = Some(Duration.ofMillis(500))))
+
+        results.expectNext(
+          2.seconds,
+          Ctx(context, Left(CompletionResponse.TimeoutResponse(commandId))),
+        )
+        succeed
+      }
 
-        // TODO(RA): test timeouts
+      "use the maximum expiry time, if provided" in {
+        val Handle(submissions, results, _, _) = runCommandTrackingFlow(
+          allSubmissionsSuccessful,
+          maximumExpiryTime = Duration.ofSeconds(1),
+        )
+
+        submissions.sendNext(submission)
+
+        results.expectNext(
+          2.seconds,
+          Ctx(context, Left(CompletionResponse.TimeoutResponse(commandId))),
+        )
+        succeed
+      }
+
+      "cap the timeout at the maximum expiry time" in {
+        val Handle(submissions, results, _, _) = runCommandTrackingFlow(
+          allSubmissionsSuccessful,
+          maximumExpiryTime = Duration.ofSeconds(1),
+        )
+
+        submissions.sendNext(newSubmission(commandId, dedupTime = Some(Duration.ofSeconds(10))))
+
+        results.expectNext(
+          2.seconds,
+          Ctx(context, Left(CompletionResponse.TimeoutResponse(commandId))),
+        )
         succeed
```

Comment:
These are good test cases! For further improving test coverage, you might want to add a test case for handling `DeduplicationPeriod.DeduplicationDuration`s as well (I understand they're handled identically to `DeduplicationPeriod.DeduplicationTime` but this is not documented by any test case).

Reply:
The code that handles the `deduplicationPeriod` will be deleted in the next PR, so I'm not so worries.